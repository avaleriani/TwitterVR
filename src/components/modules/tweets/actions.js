import moment from 'moment';
import { v4 as uuid } from 'uuid';

const scope = '@@tweets';

export const CREATE_TWEET = `${scope}/CREATE_TWEET`;

export const createTweet = payload => ({
  type: CREATE_TWEET,
  payload: {
    id: uuid(), // unique id of tweet
    createdAt: moment(), // datetime object of tweet's creation
    userId: null, // user's id
    text: '', // content of the tweet
    replyToId: null, // if the tweet is a reply, id to the original tweet
    ...payload,
  },
});
