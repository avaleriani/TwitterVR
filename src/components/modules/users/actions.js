import { v4 as uuid } from 'uuid';
import * as fromUsers from './reducer';

const scope = '@@users';

export const RECIEVE_AUTH = `${scope}/RECIEVE_AUTH`;
export const SIGNIN = `${scope}/SIGNIN`;
export const SIGNUP = `${scope}/SIGNUP`;
export const LOGIN = `${scope}/LOGIN`;
export const LOGOUT = `${scope}/LOGOUT`;

export const recieveAuth = () => ({
  type: RECIEVE_AUTH,
});

export const signin = id => ({
  type: SIGNIN,
  payload: {
    id,
  },
});

export const signup = username => ({
  type: SIGNUP,
  payload: {
    id: uuid(),
    username,
  },
});

export const login = username => (dispatch, getState) => {
  const { users } = getState();

  dispatch({ type: LOGIN, username });

  const userId = fromUsers.getUserIdByUsername(users, username);

  if (userId) {
    return dispatch(signin(userId));
  }

  return dispatch(signup(username));
};

export const logout = () => ({
  type: LOGOUT,
});
