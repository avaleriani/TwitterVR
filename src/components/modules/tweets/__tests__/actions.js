import * as actions from '../actions';

jest.mock('uuid', () => ({
  v4: () => '<id>',
}));

jest.mock('moment', () => () => '<datetime>');

describe('tweets actions', () => {
  it('exports constants', () => {
    expect(actions.CREATE_TWEET).toBeDefined;
  });

  it('creates tweet', () => {
    const tweet = actions.createTweet({
      text: 'Test tweet',
    });

    expect(tweet).toMatchSnapshot();
  });
});
