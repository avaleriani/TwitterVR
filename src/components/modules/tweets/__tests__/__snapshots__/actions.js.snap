// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tweets actions creates tweet 1`] = `
Object {
  "payload": Object {
    "createdAt": "<datetime>",
    "id": "<id>",
    "replyToId": null,
    "text": "Test tweet",
    "userId": null,
  },
  "type": "@@tweets/CREATE_TWEET",
}
`;
